components/binance/browser/binance_service.cc

```diff
@@ -0,0 +1,486 @@
+    if (headers_list) {
+      size_t iter = 0;
+      std::string key;
+      std::string value;
+      while (headers_list->EnumerateHeaderLines(&iter, &key, &value)) {
+        key = base::ToLowerASCII(key);
+        headers[key] = value;
+      }
+    }
+  }
+
+  url_loaders_.erase(iter);
+
+  std::move(callback).Run(
+      response_code, response_body ? *response_body : "", headers);
+}
+
+bool BinanceService::IsUnauthorized(int status) {
+  return status == 401;
+}
+
+bool BinanceService::SetAccessTokens(const std::string& access_token,
+                                        const std::string& refresh_token) {
+  access_token_ = access_token;
+  refresh_token_ = refresh_token;
+
+  std::string encrypted_access_token;
+  std::string encrypted_refresh_token;
+
+  if (!OSCrypt::EncryptString(access_token, &encrypted_access_token)) {
+    LOG(ERROR) << "Could not encrypt and save Binance token info";
+    return false;
+  }
+  if (!OSCrypt::EncryptString(refresh_token, &encrypted_refresh_token)) {
+    LOG(ERROR) << "Could not encrypt and save Binance token info";
+    return false;
+  }
+
+  std::string encoded_encrypted_access_token;
+  std::string encoded_encrypted_refresh_token;
+  base::Base64Encode(encrypted_access_token, &encoded_encrypted_access_token);
+  base::Base64Encode(
+      encrypted_refresh_token, &encoded_encrypted_refresh_token);
+
+  Profile* profile = Profile::FromBrowserContext(context_);
+  profile->GetPrefs()->SetString(
+      kBinanceAccessToken, encoded_encrypted_access_token);
+  profile->GetPrefs()->SetString(kBinanceRefreshToken,
+      encoded_encrypted_refresh_token);
+
+  return true;
+}
+
+bool BinanceService::LoadTokensFromPrefs() {
+  Profile* profile = Profile::FromBrowserContext(context_);
+  std::string encoded_encrypted_access_token =
+      profile->GetPrefs()->GetString(kBinanceAccessToken);
+  std::string encoded_encrypted_refresh_token =
+      profile->GetPrefs()->GetString(kBinanceRefreshToken);
+
+  std::string encrypted_access_token;
+  std::string encrypted_refresh_token;
+  if (!base::Base64Decode(encoded_encrypted_access_token,
+                          &encrypted_access_token) ||
+      !base::Base64Decode(encoded_encrypted_refresh_token,
+                          &encrypted_refresh_token)) {
+    LOG(ERROR) << "Could not Base64 decode Binance token info.";
+    return false;
+  }
+
+  if (!OSCrypt::DecryptString(encrypted_access_token, &access_token_)) {
+    LOG(ERROR) << "Could not decrypt and save Binance token info.";
+    return false;
+  }
+  if (!OSCrypt::DecryptString(encrypted_refresh_token, &refresh_token_)) {
+    LOG(ERROR) << "Could not decrypt and save Binance token info.";
+    return false;
+  }
+
+  return true;
+}
+
+std::string BinanceService::GetBinanceTLD() {
+  Profile* profile = Profile::FromBrowserContext(context_);
+
+  const std::string usTLD = "us";
+  const std::string usCode = "US";
+  const std::string globalTLD = "com";
```

Comment:
Nit: Should use underscore to fit with style guidelines.

Reply:
Fixed.